src/orion/benchmark/task/hpobench.py

```diff
@@ -0,0 +1,86 @@
+"""
+Task for HPOBench
+=================
+"""
+import importlib
+import subprocess
+from typing import Dict, List, Union
+
+from orion.algo.space.configspace import to_orionspace
+from orion.benchmark.task.base import BenchmarkTask
+
+
+class HPOBench(BenchmarkTask):
+    """Benchmark Task wrapper over HPOBench (https://github.com/automl/HPOBench)
+
+    For more information on HPOBench, see original paper at https://arxiv.org/abs/2109.06716.
+
+    Katharina Eggensperger, Philipp Müller, Neeratyoy Mallik, Matthias Feurer, René Sass, Aaron Klein,
+    Noor Awad, Marius Lindauer, Frank Hutter. "HPOBench: A Collection of Reproducible Multi-Fidelity
+    Benchmark Problems for HPO" Thirty-fifth Conference on Neural Information Processing Systems
+    Datasets and Benchmarks Track (Round 2).
+
+    Parameters
+    ----------
+    max_trials : int
+        Maximum number of trials for this task.
+    hpo_benchmark_class : str
+        Full path to a particular class of benchmark in HPOBench.
+    benchmark_kwargs: str
+        Optional parameters to create benchmark instance of class `hpo_benchmark_class`.
+    objective_function_kwargs: dict
+        Optional parameters to use when calling `objective_function` of the benchmark instance.
+    """
+
+    def __init__(
+        self,
+        max_trials: int,
+        hpo_benchmark_class: Union[str, None] = None,
+        benchmark_kwargs: dict = dict(),
+        objective_function_kwargs: dict = dict(),
+    ):
+        super().__init__(
+            max_trials=max_trials,
+            hpo_benchmark_class=hpo_benchmark_class,
+            benchmark_kwargs=benchmark_kwargs,
+            objective_function_kwargs=objective_function_kwargs,
+        )
+        self._verify_benchmark(hpo_benchmark_class)
+        self.hpo_benchmark_cls = self._load_benchmark(hpo_benchmark_class)
+        self.benchmark_kwargs = benchmark_kwargs
+        self.objective_function_kwargs = objective_function_kwargs
+
+    def call(self, **kwargs) -> List[Dict]:
+        hpo_benchmark = self.hpo_benchmark_cls(**self.benchmark_kwargs)
```

Comment:
@donglinjy Did you figure out why the singularity container is destroyed at the end of the subprocess call? If we could avoid this then we could only pay the price of building it during task instantiation.

Reply:
I think this is caused by how we now run the trials, if we create the containerized benchmark during task init, we serialize the task instance and then deserialize it as a new task instance to run, after the run complete, this new instance will be destroyed which will cause the singularity container shutdown too. Although I mentioned another possible solution at https://github.com/Epistimio/orion/pull/993#issuecomment-1253803140, which will ask some additional change to make it work in remote workers scenario.